### DISCUSSION TITLE
PWM period with min_cycle_duration - a bit confused

### DISCUSSION THREAD

**Discussion by Virgun** [ScratMan/HASmartThermostat#186] [Q&A]
Heya everyone,

I'm a bit confused by PWM and min_cycle_duration parameter. 

If I set min_cycle_duration for 5 minutes and PWM for 5 minutes it means I'll turn on only when output is 100% (so PWM is 100% = 5 min). If I'll set PWM to 15 minutes I'll turn only when output is at least 33% (so min. of 5 minutes specified by min_cycle_duration). 

And there is a question... What is the advantage of settings this to longer (or maybe shorter) period of time? If PWM of 5 minutes won't be enough it'll start next cycle until temperature is reached. With min_cycle_duration and PWM of 5 minutes for both my heater stays on between 5-60 minutes. 

pwm (Optional): Set period of the pulse width modulation. If too long, the response time of the thermostat will be too slow, leading to lower accuracy of temperature control.

If too long, the response time of the thermostat will be too slow - even with PWM of 1h and output of 1% it's only 36 seconds - which isn't really that long.

Please someone explain the PWM parameter to me like I'm 5, because I don't see any reason to set it to value higher than min_cycle_duration. Unless after calibrating PID it won't work since it takes only 100% output to get this PWD/min_cycle_duration.

**Comment by wiktor-z** [comment]
> If I set min_cycle_duration for 5 minutes and PWM for 5 minutes it means I'll turn on only when output is 100% (so PWM is 100% = 5 min). If I'll set PWM to 15 minutes I'll turn only when output is at least 33% (so min. of 5 minutes specified by min_cycle_duration).

No, based on my observations, even with control_output == 1, the heating will be activated for the duration specified by min_cycle_duration. This parameter serves as a kind of safeguard, especially for gas boilers, which may not function optimally with frequent on/off cycling. It ensures that the heating system remains active for at least the specified minimum duration, regardless of the PWM or control output value.

In my current setup, I have configured the parameters as follows:
- `pwm: 00:50:00` for easy conversion of control_output into minutes.
- `min_cycle_duration: 00:05:00` based on the recommended value found in the gas boiler's user manual, and this was also confirmed by a technician/installer.
- `min_off_cycle_duration: 00:01:00` is set to 1 minute, but I'm not entirely convinced whether this value is absolutely necessary. My gas boiler has an automation feature that shuts off heating if needed to prevent overheating.

These parameter choices are tailored to my specific setup, and it's crucial to consider the specifications and recommendations provided by the equipment manufacturer, as well as any additional features like the automatic shut-off in your gas boiler.

**Comment by ScratMan** [comment]
Hi, correct. And regarding the pwm Vs min cycle duration, since several versions, when the computed on time is shorter than the minimum cycle duration, the pwm is automatically increased to avoid overheating.
